specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/firewallPolicy.json

```diff
@@ -0,0 +1,952 @@
+          ],
+          "x-ms-enum": {
+            "name": "RuleType",
+            "modelAsString": true
+          }
+        },
+        "name": {
+          "type": "string",
+          "description": "Name of the Rule" 
+        },
+        "priority": {
+          "type": "integer",
+          "format": "int32",
+          "maximum": 65000,
+          "exclusiveMaximum": false,
+          "minimum": 100,
+          "exclusiveMinimum": false,
+          "description": "Priority of the Firewall Policy Rule resource."
+        }
+      }     
+    },
+    "FirewallPolicyNatRule": {
+      "properties": {
+        "action": {
+          "$ref": "#/definitions/FirewallPolicyNatRuleAction",
+          "description": "The action type of a Nat rule, SNAT or DNAT"
+        },
+        "translatedAddress": {
+          "type": "string",
+          "description": "The translated address for this NAT rule."
+        },
+        "translatedPort": {
+          "type": "string",
+          "description": "The translated port for this NAT rule."
+        },
+        "ruleCondition": {
+          "$ref": "#/definitions/FirewallPolicyRuleCondition",
+          "description": "The match conditions for incoming traffic"
+
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/FirewallPolicyRule"
+        }
+      ],
+      "x-ms-discriminator-value": "FirewallPolicyNatRule",
+      "description": "Firewall Policy NAT Rule"
+    },
+    "FirewallPolicyFilterRule": {
+      "properties": {
+        "action": {
+          "$ref": "#/definitions/FirewallPolicyFilterRuleAction",
+          "description": "The action type of a Filter rule"
+        },
+        "ruleConditions": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/FirewallPolicyRuleCondition"
+          },
+          "description": "Collection of rule conditions used by a rule."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/FirewallPolicyRule"
+        }
+      ],
+      "x-ms-discriminator-value": "FirewallPolicyFilterRule",
+      "description": "Firewall Policy Filter Rule"
+    },
+    "FirewallPolicyRuleCondition": {
+      "description": "Properties of a rule.",
+      "discriminator": "ruleConditionType",
+      "required": [
+        "ruleConditionType"
+      ],
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Name of the rule condition."
+        },
+        "description": {
+          "type": "string",
+          "description": "Description of the rule condition."
+        },
+        "ruleConditionType": {
+          "type": "string",
+          "description": "RuleConditionType - ApplicationRuleCondition or NetworkRuleCondition.",
+          "enum": [
+            "ApplicationRuleCondition",
+            "NetworkRuleCondition"
+          ],
+          "x-ms-enum": {
+            "name": "RuleConditionType",
```

Comment:
Since all network's enums exist in one namespace please rename this one to something like `FirewallPolicyRuleConditionType` to avoid name collision and make it more clear what this enum represents